ndc-reference/tests/capabilities/expected.json

```diff
@@ -3,7 +3,11 @@
   "capabilities": {
     "query": {
       "aggregates": {},
-      "variables": {}
+      "variables": {},
+      "nested_field_capabilities": {
```

Comment:
Let's call this `nested_fields` please, we don't repeat `_capabilities` elsewhere. Same for below, I think, I think we can just call the sub-capabilities `filter_by` and `order_by` since they are scoped by inclusion.

Also, I think we should either make `nested_field_capabilities` required, since it has no meaning on its own, or lift `query.nested_fields.filter_by` to `query.filter_by_nested_fields` etc.

We also need to document these new capabilities in the spec, on the two new pages, and on the capabilities page, please.